Source/Core/DolphinQt/SkylanderPortal/SkylanderPortalWindow.cpp

```diff
@@ -92,71 +242,268 @@ void SkylanderPortalWindow::CreateMainWindow()
+    dir += QDir::separator();
+    m_path_edit->setText(dir);
+    m_collection_path = dir;
+  }
+  Config::SetBase(Config::MAIN_SKYLANDERS_PATH, dir.toStdString());
+
+  if (m_only_show_collection->isChecked())
+    RefreshList();
+}
+
+void SkylanderPortalWindow::LoadSelected()
+{
+  if (!m_emulating)
+    return;
+
+  u8 slot = GetCurrentSlot();
 
-  auto* combo_skylist = new QComboBox();
-  QStringList filterlist;
-  for (const auto& entry : IOS::HLE::USB::list_skylanders)
+  QDir collection = QDir(m_collection_path);
+  QString skyName;
+  QString file_path;
+  if (m_only_show_collection->isChecked())
   {
-    const uint qvar = (entry.first.first << 16) | entry.first.second;
-    combo_skylist->addItem(QString::fromStdString(entry.second), QVariant(qvar));
-    filterlist << QString::fromStdString(entry.second);
+    if (m_skylander_list->currentItem() == nullptr)
+      return;
+    file_path = m_collection_path + m_skylander_list->currentItem()->text() + tr(".sky");
+  }
+  else
+  {
+    file_path = GetFilePath(sky_id, sky_var);
   }
-  combo_skylist->addItem(tr("--Unknown--"), QVariant(0xFFFFFFFF));
-  combo_skylist->setEditable(true);
-  combo_skylist->setInsertPolicy(QComboBox::NoInsert);
 
-  auto* co_compl = new QCompleter(filterlist, this);
-  co_compl->setCaseSensitivity(Qt::CaseInsensitive);
-  co_compl->setCompletionMode(QCompleter::PopupCompletion);
-  co_compl->setFilterMode(Qt::MatchContains);
-  combo_skylist->setCompleter(co_compl);
+  if (file_path.isEmpty())
+  {
+    QMessageBox::StandardButton createFileResponse;
+    createFileResponse =
+        QMessageBox::question(this, tr("Create Skylander File"),
+                              tr("Skylander not found in this collection. Create new file?"),
+                              QMessageBox::Yes | QMessageBox::No);
 
-  layout->addWidget(combo_skylist);
+    if (createFileResponse == QMessageBox::Yes)
+    {
+      QString predef_name = m_collection_path;
+      const auto found_sky = IOS::HLE::USB::list_skylanders.find(std::make_pair(sky_id, sky_var));
+      if (found_sky != IOS::HLE::USB::list_skylanders.end())
+      {
+        predef_name += QString::fromStdString(std::string(found_sky->second.name) + ".sky");
+      }
+      else
+      {
+        QString str = tr("Unknown(%1 %2).sky");
+        predef_name += str.arg(sky_id, sky_var);
+      }
+
+      CreateSkyfile(predef_name, true);
+    }
+  }
+  else
+  {
+    m_last_skylander_path = QFileInfo(file_path).absolutePath() + tr("/");
 
-  auto* line = new QFrame();
-  line->setFrameShape(QFrame::HLine);
-  line->setFrameShadow(QFrame::Sunken);
-  layout->addWidget(line);
+    LoadSkyfilePath(slot, file_path);
+  }
+}
+
+void SkylanderPortalWindow::LoadFromFile()
+{
+  u8 slot = GetCurrentSlot();
+  const QString file_path =
+      DolphinFileDialog::getOpenFileName(this, tr("Select Skylander File"), m_last_skylander_path,
+                                         QString::fromStdString("Skylander (*.sky);;"));
```

Comment:
The two trailing `;;` cause an empty filter in macOS, not sure about other OS's. 

```cpp
QString::fromStdString("Skylander (*.sky);;All Files (*)"));
```
should suffice. You may also want to convert `QString::fromStdString` to `tr` for use with translations.